grapher/stackedCharts/MarimekkoChart.tsx

```diff
@@ -782,29 +792,46 @@ export class MarimekkoChart
     @computed private get sortedItems(): Item[] {
         const { items, sortConfig } = this
 
-        let sortByFunc: (item: Item) => number | string | undefined
+        let sortFunc: (a: Item, b: Item) => number
         switch (sortConfig.sortBy) {
             case SortBy.custom:
-                sortByFunc = () => undefined
+                sortFunc = (): number => 0
                 break
             case SortBy.entityName:
-                sortByFunc = (item: Item): string => item.entityName
+                sortFunc = (a: Item, b: Item): number =>
+                    a.entityName.localeCompare(b.entityName)
                 break
             case SortBy.column:
                 const sortColumnSlug = sortConfig.sortColumnSlug
-                sortByFunc = (item: Item): number =>
-                    item.bars.find((b) => b.seriesName === sortColumnSlug)
-                        ?.yPoint.value ?? 0
+                sortFunc = (a: Item, b: Item): number => {
+                    const aValue =
+                        a.bars.find((bar) => bar.seriesName === sortColumnSlug)
+                            ?.yPoint.value ?? 0
+                    const bValue =
+                        b.bars.find((bar) => bar.seriesName === sortColumnSlug)
+                            ?.yPoint.value ?? 0
+                    const diff = aValue - bValue
+                    if (diff !== 0) return diff
+                    return a.entityName.localeCompare(b.entityName)
+                }
                 break
             default:
             case SortBy.total:
-                sortByFunc = (item: Item): number => {
-                    const lastPoint = last(item.bars)?.yPoint
-                    if (!lastPoint) return 0
-                    return lastPoint.valueOffset + lastPoint.value
+                sortFunc = (a: Item, b: Item): number => {
+                    const aLastPoint = last(a.bars)?.yPoint
+                    const bLastPoint = last(b.bars)?.yPoint
+                    const aValue =
+                        (aLastPoint?.valueOffset ?? 0) +
+                        (aLastPoint?.value ?? 0)
+                    const bValue =
+                        (bLastPoint?.valueOffset ?? 0) +
+                        (bLastPoint?.value ?? 0)
+                    const diff = aValue - bValue
+                    if (diff !== 0) return diff
+                    else return a.entityName.localeCompare(b.entityName)
                 }
         }
-        const sortedItems = sortBy(items, sortByFunc)
+        const sortedItems = items.sort(sortFunc)
```

Comment:
`.sort()` sorts in place, which is potentially fine for this context, but I think it's better to avoid it to prevent any hard-to-debug bugs.